elastic/kibana
# [ObsUx][Infra] Add collapsible sections in the overview tab

Closes #175989

## Summary

This PR is a follow-up to https://github.com/elastic/kibana/issues/175558. It adds the active alerts count next to the alert section title (this will happen after the alerts widget is loaded) following the rules:

Default behaviour
No alerts at all ==> Collapse and say 'No active alerts'
No active alerts ==> Collapse and say 'No active alerts'
Active alerts ==> Expand fully

Collapsed
No alerts at all ==> Say 'No active alerts'
No active alerts ==> Say 'No active alerts'
Active alerts ==> say "X active alerts"


It adds a change in the `AlertSummaryWidget` to make it possible to get the alerts count after the widget is loaded using a new prop.

This PR also changes the alerts tab active alert count badge color on the hosts view to keep it consistent:

| Before | After |
| ------ | ------ |
| <img width="242" alt="image" src="https://github.com/elastic/kibana/assets/14139027/85beec43-6522-4d58-a808-d3f7ec3e0759"> | <img width="263" alt="image" src="https://github.com/elastic/kibana/assets/14139027/43983493-3270-471a-8788-40ce38bed334"> |

## Testing
- Open hosts view and select a host with active alerts (flyout or full page)
   - The alerts section should be expanded showing the alerts widget
      <img width="1920" alt="image" src="https://github.com/elastic/kibana/assets/14139027/f851c914-96cf-475f-bab3-dddc485405ec">fd1a21035a5f)
   - Collapse the alerts section by clicking on the title or the button:
     <img width="1917" alt="image" src="https://github.com/elastic/kibana/assets/14139027/e23e09ed-a781-4961-b592-6f13f539c316">
-  Open hosts view and select a host without active alerts (flyout or full page)
   - The alerts section should be collapsed showing the message 'No active alerts'
    ![Image](https://github.com/elastic/obs-infraobs-team/assets/14139027/7077d3b3-c020-4be5-a3da-b46dda0d3ae0)

https://github.com/elastic/kibana/assets/14139027/4058ed69-95f5-4b4c-8925-6680ac3791c1
